src/components/Carousel.tsx

```diff
@@ -1,18 +1,186 @@
-import React from 'react';
+import { Component } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+interface Props {
+  images: string[];
+}
+
+interface State {
+  position: number;
+  frameSize: number
+  step: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+}
+
+export class Carousel extends Component<Props, State> {
+  state = {
+    position: 0,
+    frameSize: 3,
+    step: 3,
+    itemWidth: 130,
+    animationDuration: 1,
+    infinite: false,
+  };
+
+  render() {
+    const {
+      position,
+      step,
+      itemWidth,
+      frameSize,
+      animationDuration,
+      infinite,
+    } = this.state;
+    const { images } = this.props;
+
+    const handleNextClick = () => {
+      if (infinite && position === images.length - frameSize) {
+        this.setState({
+          position: 0,
+        });
+      } else {
+        this.setState({
+          position: Math.min(position + step, images.length - frameSize),
+        });
+      }
+    };
+
+    const handlePrevClick = () => {
+      if (infinite && position === 0) {
+        this.setState({
+          position: images.length - frameSize,
+        });
+      } else {
+        this.setState({ position: Math.max(position - step, 0) });
+      }
+    };
+
+    return (
+      <div className="Carousel">
+        <div className="Carousel__container">
+          <button
+            disabled={position === 0 && !infinite}
+            className="Button"
+            type="button"
+            onClick={handlePrevClick}
+          >
+            {'<'}
+          </button>
+
+          <div
+            className="Carousel__list-container"
+            style={
+              {
+                width: `${itemWidth * frameSize}px`,
+              }
+            }
+          >
+            <ul
+              className="Carousel__list"
+              style={
+                {
+                  transform: `translateX(-${position * itemWidth}px)`,
+                  transition: `transform ${animationDuration}s`,
+                }
+              }
+            >
+              {images.map((image, index) => (
+                <li key={image}>
+                  <img
+                    src={image}
+                    alt={`${index + 1}`}
```

Comment:
the `alt` attribute should give us the alternative info. I understand it is not easy in this case, but let's try to come up with sth like that
```suggestion
                    alt={`emoji ${index + 1}`}
```